aggregator/src/aggregator/collection_job_driver.rs

```diff
@@ -76,6 +81,7 @@ where
             metrics: CollectionJobDriverMetrics::new(meter),
             batch_aggregation_shard_count,
             collection_retry_strategy,
+            max_future_concurrency: 1000,
```

Comment:
I will wire this up in the configuration file once we agree that this is the right way to go.